framework/src/modules/chain/submodules/loader.js

```diff
@@ -755,8 +755,18 @@ __private.createSnapshot = height => {
 	}
 
 	const snapshotRound = library.config.loading.snapshotRound;
+	// Negative number not possible as `commander` does not recognize this as valid flag (throws error)
+	if (
+		(!snapshotRound && Number.isNaN(parseInt(snapshotRound))) ||
+		parseInt(snapshotRound) < 0
+	) {
+		throw new Error(
+			'Unable to create snapshot, "--snapshot" parameter should be an integer equal to or greater than zero'
```

Comment:
> parameter should be an integer equal to or greater than zero

This definition does not apply to examples you mentioned in the PR. So you can actually pass snapshot parameter without any value and it will be considered as all rounds. Please fix this comment or enforce user to pass `0` incase he wants to snapshot all rounds.